matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -209,18 +218,47 @@ pub enum PeerState {
     /// - The peer left the signalling server
     Disconnected,
 }
+/// Used to send and recieve packets on a given web rtc channel
+#[derive(Debug)]
+pub struct WebRtcChannel {
+    tx: UnboundedSender<(PeerId, Packet)>,
+    rx: UnboundedReceiver<(PeerId, Packet)>,
+}
 
-/// Contains the interface end of a full-mesh web rtc connection
-///
-/// Used to send and receive messages from other peers
+impl WebRtcChannel {
+    /// Call this where you want to handle new received messages from a specific channel as
+    /// configured in [`WebRtcSocketBuilder::channels`]. The index of a channel is its index in
+    /// the vec [`WebRtcSocketBuilder::channels`] as you configured it before (or 0 for the
+    /// default channel if you use the default configuration).
+    ///
+    /// messages are removed from the socket when called
+    pub fn receive(&mut self) -> Vec<(PeerId, Packet)> {
+        std::iter::repeat_with(|| self.rx.try_next())
+            .map_while(Result::ok)
+            .flatten()
+            .collect()
+    }
+
+    /// Send a packet to the given peer on a specific channel as configured in
+    /// [`WebRtcSocketBuilder::channels`].
+    ///
+    /// The index of a channel is its index in the vec [`WebRtcSocketBuilder::channels`] as you
+    /// configured it before (or 0 for the default channel if you use the default
+    /// configuration).
+    pub fn send(&mut self, packet: Packet, peer: PeerId) {
+        self.tx.unbounded_send((peer, packet)).expect("Send failed");
+    }
+}
+
+/// Contains a set of web rtc channels and some connection metadata.
 #[derive(Debug)]
+#[cfg_attr(feature = "bevy-plugin", derive(Resource))]
```

Comment:
Why not just add a

```rust
#[derive(Resource, Deref, DerefMut, From)]
pub struct MatchboxSocket(WebRtcSocket);
```

to the bevy plugin?

Reply:
Largely because I forgot bevy had `DerefMut`, will change it to this

Reply:
We don't currently depend on `derive_more` so will leave out `From` for now